File: caesar.c
#include <stdio.h>
#include <stdlib.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main (int argc, string argv[1])
{
    //check for 2 arguments only and return from main a value of 1 if error
    if (argc != 2)
    {
        printf("Invalid input.  Try again. \n");
        return 1;
    }
    
    else 
    {
        printf("Text to encrypt: \n");
    }
    
    //once correct, put key into k
    int k = atoi(argv[1]);
    
    string p = get_string();
    
    //convert text to cipher
        printf("ciphertext: ");
        for (int i = 0, n = strlen(p); i < n; i++)
        {
           
            //check case of letter and then convert
            if (islower(p[i]))
                printf("%c", (((p[i] + k) - 97) % 26) + 97);
            else if (isupper(p[i]))
                printf("%c", (((p[i] + k) - 65) % 26) + 65);
            else
                printf("%c", p[i]);
        }

    //After outputting ciphertext, you should print a newline. Your program should then exit by returning 0 from main.
    printf("\n");
    return 0;
}

File: caesar.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("Invalid input, try again: ")
        exit(1)
    
    print("plaintext: ", end = "")
        
    key = int(sys.argv[1])
    ciphertext = []
    plaintext = cs50.get_string()
    for symbol in plaintext:
        if symbol.isalpha():
            ciphertext.append(caesar(symbol, key))
        else:
            ciphertext.append(symbol)
                
    print("ciphertext: ", "".join(ciphertext))
    exit(0)

def caesar(char, key):
    if char.isupper():
        return chr(((ord(char) - 65 + key) % 26) + 65)
    else:
        return chr(((ord(char) - 97 + key) % 26) + 97)
        
if __name__ == "__main__":
    main()